### PR TITLE
Jekrame/unit testing

### PR DESCRIPTION
**Pull request by JKrame**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ x ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
follow instructions in Dev_Readme.md to set up the repo
```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code
Run command `npm test`
```

## What to Check
Verify that the following are valid
* ...

## Other Information
<!-- Add any other helpful information that may be needed here. -->

### CODE REVIEW

File: src/json.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+var json = {
```
Comment (gogowings) [inline src/json.js]: nitpick: can we make it a bit specific than json.js?

File: src/widget.js [hunk 2]
```diff
@@ -18,67 +22,70 @@ var widget = {
...
-                keyword = parentNode.dataset.browsetopic;
-            }
+                if(key !== "search")
+                {
+                    keyword = (key!="all") ? key : parentNode.dataset.browsetopic;
+                    
+                    parentNode.setAttribute("id", widget_name);
+                    widget.appendToWidget("#" + widget_name, "div", "", '<div class="gh-widget-container"><div class="gh-widget-item gh-widget-photo"></div><div class="gh-widget-item gh-widget-personal-details"></div></div><div class="gh-widget-container gh-widget-stats"></div><hr class="gh-widget-hr"><div class="gh-widget-container"><div class="gh-widget-item gh-widget-heading">Top repositories for "' + keyword + '"</div></div><div class="gh-widget-repositories"></div><div class="gh-widget-container"><div class="gh-widget-item gh-widget-follow"></div><div class="gh-widget-item gh-widget-active-time"></div></div>')
+                    widget.fetchRepos(keyword, "#" + widget_name, orgs);
+                }
```
Comment (gogowings) [inline src/widget.js:71]: seems like quite some duplicates between line 51-53 and line 62-64. can we abstract the common code?

File: src/widget.js [hunk 3]
```diff
@@ -18,67 +22,70 @@ var widget = {
...
             }
-            else
+            else if (type === "browse")
             {
-                keyword = parentNode.dataset.browsetopic;
-            }
+                if(key !== "search")
+                {
+                    keyword = (key!="all") ? key : parentNode.dataset.browsetopic;
+
```
Comment (gogowings) [inline src/widget.js]: when type=="browse", what scenarios will cause key == 'search' or 'all'?

Comment (JKrame) [reply src/widget.js]: key is "search" when the search button is called, it is "all" upon initialization, this is to detect which widget is calling the event so we can target the correct widget without reloading both

File: src/widget.js (hunk 2 above)
Comment (JKrame) [reply src/widget.js:71]: Yes, these lines have been abstracted out and I have pushed the skipping implementation we discussed in my office

File: src/json.js (hunk 1 above)
Comment (JKrame) [reply src/json.js]: Renamed to githubApiInterface

File: src/widget.js [hunk 4]
```diff
@@ -40,32 +40,39 @@ var widget = {
...
             var widget_name =  type + "_widget" + i;
-            
             var keyword = '';
+
+            /* 
+                If key is search or all, update both widgets, otherwise update one and skip the other since this
+                iterates all widgets every time a click event is fired. Not continuing would cause both widgets to
+                reload on any click, creating a bad user experience. 
+            */
+
```
Comment (gogowings) [inline src/widget.js:50]: Good! I was about to add a comment that we will generally need more code comments in your PR :) :+1:

File: src/widget.js (hunk 2 above)
Comment (gogowings) [reply src/widget.js:71]: :+1:

**Review by gogowings — APPROVED** [review]

**Pull request merged**